lms/djangoapps/instructor_task/tests/test_tasks_helper.py

```diff
@@ -734,6 +734,26 @@ def test_problem_grade_report(self):
             ))
         ])
 
+    def test_problem_grade_report_valid_columns_order(self):
+        """
+        Test that in the CSV grade report columns are placed in the proper order
+        """
+        problem_names = [u'Homework 1: Problem - pröblem_a_url']
+        for i in xrange(30):
+            problem_url_a = 'test_problem_a_%d' % i
+            self.define_option_problem(problem_url_a, parent=self.vertical_a)
+            title = 'Homework 1: Problem - %s' % problem_url_a
+            problem_names.append(title)
+        problem_names.append(u'Homework 1: Problem - pröblem_b_url')
+
+        header_row = [u'Student ID', u'Email', u'Username', u'Final Grade']
+        for problem in problem_names:
+            header_row += [problem + ' (Earned)', problem + ' (Possible)']
+
+        with patch('instructor_task.tasks_helper._get_current_task'):
+            upload_problem_grade_report(None, None, self.course.id, None, 'graded')
+        self.assertEquals(self.get_csv_row_with_headers(), header_row)
```

Comment:
I don't understand how this test tests that the problems are appearing in the correct order.